iphone/TitaniumKit/TitaniumKit/Sources/API/TiUIView.m

```diff
@@ -644,19 +647,82 @@ - (void)setBackgroundTopCap_:(id)value
   }
 }
 
-- (void)setBorderRadius_:(id)radius
+- (CGFloat)radiusFromObject:(id)object
 {
-  TiDimension theDim = TiDimensionFromObject(radius);
-  if (TiDimensionIsDip(theDim)) {
-    self.layer.cornerRadius = MAX(theDim.value, 0);
+  TiDimension theDim = TiDimensionFromObject(object);
+  return TiDimensionIsDip(theDim) ? MAX(theDim.value, 0) : 0;
+}
+
+- (void)addCornerRadius:(NSArray *)radiusArray toLayer:(CALayer *)viewLayer
+{
+  CGFloat topLeftRadius;
+  CGFloat bottomLeftRadius;
+  CGFloat topRightRadius;
+  CGFloat bottomRightRadius;
+
+  if (radiusArray.count >= 4) {
+    topLeftRadius = [self radiusFromObject:radiusArray[0]];
+    topRightRadius = [self radiusFromObject:radiusArray[1]];
+    bottomRightRadius = [self radiusFromObject:radiusArray[2]];
+    bottomLeftRadius = [self radiusFromObject:radiusArray[3]];
+  } else if (radiusArray.count >= 2) {
+    CGFloat radius = [self radiusFromObject:radiusArray[0]];
+    topLeftRadius = radius;
+    bottomRightRadius = radius;
+
+    radius = [self radiusFromObject:radiusArray[1]];
+    bottomLeftRadius = radius;
+    topRightRadius = radius;
+  } else if (radiusArray.count == 1) {
+    // For same corner radius, no need to create bezier path. Use CALayer's cornerRadius.
+    viewLayer.cornerRadius = [self radiusFromObject:radiusArray[0]];
+    return;
+  }
+
+  CAShapeLayer *shapeLayer = [[CAShapeLayer alloc] initWithLayer:viewLayer];
+  CGRect rect = viewLayer.bounds;
```

Comment:
I don't see any other usage of `rect` apart from `rect.size`, so why not directly pull the size and use only that? Avoids having to type `rect.size.*` again and again.

```suggestion
  CGSize size = viewLayer.bounds.size;
```

Reply:
Agree. Updated!